### DISCUSSION TITLE
VPN + OpenAI gives "Error updating user info"

### DISCUSSION THREAD

**Discussion by lxp731** [RamiAwar/dataline#285] [Q&A] [answered]
I ran "docker run -p 7377:7377 -v dataline:/home/.dataline --name dataline ramiawar/dataline:latest" directly and then opened http:// 127.0.0.7:7377, input OPEN_API_KEY, finally there is "error updating user info", what should I do?

**Comment by RamiAwar** [comment]
Hey @lxp731 ! Can you check the docker logs after you try this and send them over? We'll have more info about what's going wrong from there.

This error might mean a lot of things, but usually it's an issue with setting up the OpenAI API connection.
An inability to connect to openai servers, or maybe a lack of openai credits, or maybe something else.

You can see the docker logs by opening the docker dashboard and pressing on the dataline container then going to Logs.

**Reply by lxp731** [reply]
### The Logs

Create Date: 2024-07-11 23:28:01.641837
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:7377 (Press CTRL+C to quit)
INFO:     172.20.0.1:45858 - "GET / HTTP/1.1" 200 OK
INFO:     172.20.0.1:45868 - "GET /index-Dkn2NR9y.js HTTP/1.1" 200 OK
INFO:     172.20.0.1:45858 - "GET /index-Bgb_Eakv.css HTTP/1.1" 200 OK
INFO:     172.20.0.1:45884 - "GET /healthcheck HTTP/1.1" 200 OK
INFO:     172.20.0.1:45900 - "HEAD /auth/login HTTP/1.1" 405 Method Not Allowed
INFO:     172.20.0.1:45900 - "GET /healthcheck HTTP/1.1" 200 OK
INFO:     172.20.0.1:45904 - "GET /connections HTTP/1.1" 200 OK
INFO:     172.20.0.1:45914 - "GET /conversations HTTP/1.1" 200 OK
INFO:     172.20.0.1:45884 - "GET /settings/info HTTP/1.1" 404 Not Found
INFO:     172.20.0.1:45884 - "GET /healthcheck HTTP/1.1" 200 OK
INFO:     172.20.0.1:45884 - "GET /healthcheck HTTP/1.1" 200 OK
INFO:     172.20.0.1:45884 - "OPTIONS /settings/info HTTP/1.1" 200 OK
INFO:openai._base_client:Retrying request to /models in 0.823672 seconds
INFO:openai._base_client:Retrying request to /models in 1.575162 seconds
INFO:     172.20.0.1:45914 - "GET /healthcheck HTTP/1.1" 200 OK
ERROR:dataline.app:Connection error.
NoneType: None
INFO:     172.20.0.1:45884 - "PATCH /settings/info HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):

**Reply by RamiAwar** [reply]
It indeed looks like requests to OpenAI are failing.

Are you in China? If so, are you using a VPN to access OpenAI?

I know they recently banned access 😠 That complicates things a little.

@anthony2261 is working on supporting other providers + open source models as we speak though. Should be out in a few weeks!

**Reply by lxp731** [reply]
Yes, you're right, I use a VPN for access, but I've had success deploying once in a while on the Windows version, and not once on the Linux versions (docker and Liunx). From the analysis so far it can be assumed that this is a network issue. Looking forward to your subsequent just updates.

**Reply by lxp731** [reply]
I saw that someone else had the same problem, and after a few days of fiddling around I seem to have solved it, and I'm giving you feedback today, first of all I'm using a Linux source package, and as far as I can tell all my attempts at Docker were unsuccessful. I used the tool "clash verge:V1.7.3" and had to look for a node with normal access to openAI, and in the browser with normal access to the official website of openAI. My OS is Ubuntu: 24.04, don't use the Ubuntu Shell, I don't know why, it seems to block the proxy. I am using WindTerm the Shell, configure this Shell agent, source start dataline,clash verge open TUN mode and LAN connection.

I hope to inspire you!

![image](https://github.com/user-attachments/assets/aff2c417-dedf-419a-8960-bbf760fdb488)
![image](https://github.com/user-attachments/assets/104bf4a6-b1c1-478c-aae1-54b54bec9917)
![image](https://github.com/user-attachments/assets/4a871f10-b9b2-47f4-8bb1-3fafad3412b5)

**Reply by anthony2261** [reply, marked as answer]
Super grateful for you sharing your solution🙏 !

We've just merged branch #287 which adds the ability to set a base_url in the app itself. You can stick to your solution or use that new setting once the new release is out!
